src/IonBinaryReader.ts

```diff
@@ -138,16 +141,23 @@ export class BinaryReader implements Reader {
     return this._parser.hasAnnotations();
   }
 
-  annotations() : string[] {//TODO binary support
-      return this._parser.getAnnotations();
+  private _loadAnnotations() : void {
+    if (this._annotations === null) {
+      this._annotations = [];
+      this._parser.getAnnotations().forEach(id => {
+        this._annotations.push(this.getSymbolString(id));
+      });
+    }
+  }
+
+  annotations() : string[] {
+    this._loadAnnotations();
+    return this._annotations;
   }
 
   getAnnotation(index: number) : string {
```

Comment:
I'm surprised this method exists since `annotations()` above provides direct access to the array itself.